crypto/property/property.c

```diff
@@ -649,13 +651,21 @@ static void impl_cache_flush_one_alg(ossl_uintmax_t idx, ALGORITHM *alg,
 static void ossl_method_cache_flush_some(OSSL_METHOD_STORE *store)
 {
     IMPL_CACHE_FLUSH state;
+    static TSAN_QUALIFIER uint32_t global_seed = 1;
 
     state.nelem = 0;
-    if ((state.seed = OPENSSL_rdtsc()) == 0)
-        state.seed = 1;
+    state.using_global_seed = 0;
+    if ((state.seed = OPENSSL_rdtsc()) == 0) {
+        /* If there is no timer available, seed another way */
+        state.using_global_seed = 1;
+        state.seed = tsan_load(&global_seed);
+    }
     store->cache_need_flush = 0;
     ossl_sa_ALGORITHM_doall_arg(store->algs, &impl_cache_flush_one_alg, &state);
     store->cache_nelem = state.nelem;
+    /* Without a timer, update the global seed */
+    if (state.using_global_seed)
+        tsan_add(&global_seed, state.seed);
```

Comment:
This seems to just double the global_seed every time (barring multiple threads racing with each other). So with a `uint32_t` that only gives 32 possible seed values. So, only, very slightly less deterministic. Wouldn't a simple increment be better?

Reply:
It doesn't do anything remotely like that: state.seed is updated many times on line 633 as part of the doall call.  This continues the pseudo random sequence from where it left off.  At least single threaded...

Reply:
Ah! the update on line 633 was the bit I was missing.Thanks.